### ISSUE TITLE
Bake resolves with rez<3.2.0 can't be loaded due to missing `testing` key

### ISSUE THREAD

**Issue by samson-jerome** [AcademySoftwareFoundation/rez#1862]
Baked resolves generated with older versions are not compatible with rez-3.2.0.
Note a regression per say but a breaking change 

**Environment**
* windows 10
* 3.2.0
* Python 3.11.4

**To Reproduce**
1. using rez-3.0.0, bake a simple resolve into a rxt file
```
> rez --version
Rez 3.0.0 from c:\rez\3.0.0\Lib\site-packages\rez (python 3.11)
> rez env python --output py.rxt
```

2. using rez-3.2.0, resolve a context from the above rxt file
```
> rez env --input py.rxt
12:10:10 ERROR    ResolvedContextError: Failed to load context from py.rxt: KeyError: 'testing'
```

**Expected behavior**
We should be left in a resolved context

**Actual behavior**
Resolution failure.

**Related Issues/PRs**
* #1740

**Comment by JeanChristopheMorinPerso** [comment]
Thanks for opening this. It's definitely a bug.

**Comment by JeanChristopheMorinPerso** [comment]
A fixed has been released in rez 3.2.1. Thanks again for reporting the problem.
